### PR TITLE
Add CI workflows

### PR DESCRIPTION
**Pull request by IzumiSy**
Add two workflows for all packages and examples:

* Lint
* Type checking
* Build

### CODE REVIEW

File: .github/workflows/ci-examples.yaml
Comment (IzumiSy) [inline .github/workflows/ci-examples.yaml:1]: In examples, we have only todoapp here. I will make this workflow reusable if we have more examples in future.

Comment (toiroakr) [reply .github/workflows/ci-examples.yaml:1]: In this workflow, a build job is executed for all packages and examples when one package or example is changed.
It is better to limit the execution of the build job to changed directories.
ref. https://tech-blog.yayoi-kk.co.jp/entry/2022/05/10/110000

Comment (IzumiSy) [reply .github/workflows/ci-examples.yaml:1]: @toiroakr 
I have made update: use matrix and tj-actions/changed-files and merged workflows into two: `ci-packages` and `ci-examples`. `ci-examples` is a workflow that is simply triggerd by changes in each example app, so nothing else to say about this. 

On the other hand, `ci-packages` has two jobs. The first one is to lint and type-check the package that is changed in the last commit by using changed-files action. The following job is to build all example apps. I would like to leave dependency check (what changes in packages affect example apps) to turborepo so I decided to run build on the all apps. I think I could use cache mechanism in turborepo for efficiency, but we now still don't have any problem about time, so we can put it off.

**Review by toiroakr — APPROVED** [review]
LGTM

As for the matrix, you should be able to avoid managing the list yourself by creating it dynamically from the changed file.
I'll make a pull request when I have time.

**Comment by IzumiSy** [comment]
Thanks!

**Pull request merged**